src/System.Runtime/tests/Performance/Perf.Double.cs

```diff
@@ -2,20 +2,370 @@
+        [InlineData("G4", double.NaN)]
+        [InlineData("G4", double.PositiveInfinity)]
+        [InlineData("G4", double.NegativeInfinity)]
+        [InlineData("G4", 0xFFFFFFFFFFFFF)]
+        [InlineData("G4", 0x800FFFFFFFFFFFFF)]
+        [InlineData("G4", 0x7FFFFFFFFFFFF)]
+        [InlineData("G4", 0x8000000000000000)]
+        [InlineData("G4", 0x0000000000000000)]
+        [InlineData("G3", 104234.343)]
+        [InlineData("G3", double.MinValue / 2)]
+        [InlineData("G3", Math.PI)]
+        [InlineData("G3", Math.E)]
+        [InlineData("G3", double.MaxValue)]
+        [InlineData("G3", double.MinValue)]
+        [InlineData("G3", double.NaN)]
+        [InlineData("G3", double.PositiveInfinity)]
+        [InlineData("G3", double.NegativeInfinity)]
+        [InlineData("G3", 0xFFFFFFFFFFFFF)]
+        [InlineData("G3", 0x800FFFFFFFFFFFFF)]
+        [InlineData("G3", 0x7FFFFFFFFFFFF)]
+        [InlineData("G3", 0x8000000000000000)]
+        [InlineData("G3", 0x0000000000000000)]
+        [InlineData("G2", 104234.343)]
+        [InlineData("G2", double.MinValue / 2)]
+        [InlineData("G2", Math.PI)]
+        [InlineData("G2", Math.E)]
+        [InlineData("G2", double.MaxValue)]
+        [InlineData("G2", double.MinValue)]
+        [InlineData("G2", double.NaN)]
+        [InlineData("G2", double.PositiveInfinity)]
+        [InlineData("G2", double.NegativeInfinity)]
+        [InlineData("G2", 0xFFFFFFFFFFFFF)]
+        [InlineData("G2", 0x800FFFFFFFFFFFFF)]
+        [InlineData("G2", 0x7FFFFFFFFFFFF)]
+        [InlineData("G2", 0x8000000000000000)]
+        [InlineData("G2", 0x0000000000000000)]
+        [InlineData("G1", 104234.343)]
+        [InlineData("G1", double.MinValue / 2)]
+        [InlineData("G1", Math.PI)]
+        [InlineData("G1", Math.E)]
+        [InlineData("G1", double.MaxValue)]
+        [InlineData("G1", double.MinValue)]
+        [InlineData("G1", double.NaN)]
+        [InlineData("G1", double.PositiveInfinity)]
+        [InlineData("G1", double.NegativeInfinity)]
+        [InlineData("G1", 0xFFFFFFFFFFFFF)]
+        [InlineData("G1", 0x800FFFFFFFFFFFFF)]
+        [InlineData("G1", 0x7FFFFFFFFFFFF)]
+        [InlineData("G1", 0x8000000000000000)]
+        [InlineData("G1", 0x0000000000000000)]
+        [InlineData("G0", 104234.343)]
+        [InlineData("G0", double.MinValue / 2)]
+        [InlineData("G0", Math.PI)]
+        [InlineData("G0", Math.E)]
+        [InlineData("G0", double.MaxValue)]
+        [InlineData("G0", double.MinValue)]
+        [InlineData("G0", double.NaN)]
+        [InlineData("G0", double.PositiveInfinity)]
+        [InlineData("G0", double.NegativeInfinity)]
+        [InlineData("G0", 0xFFFFFFFFFFFFF)]
+        [InlineData("G0", 0x800FFFFFFFFFFFFF)]
+        [InlineData("G0", 0x7FFFFFFFFFFFF)]
+        [InlineData("G0", 0x8000000000000000)]
+        [InlineData("G0", 0x0000000000000000)]
+        public void ToStringWithFormat(string format, double number)
```

Comment:
I would also test `F7`, `F9`, `F15`, `F17`, and `F50` as well as just `F` and just `E`.

`G` picks the more compact of either `F` (fixed-point) or `E` (exponential).

As for the precisions given above, `7`, `9`, `15`, and `17` are all values we have special paths for and `50` is the maximum number of supported digits for the `DoubleToNumber` native method.